src/Engine/Scene/SkeletonBasedAnimationSystem.cpp

```diff
@@ -0,0 +1,149 @@
+#include <Engine/Scene/SkeletonBasedAnimationSystem.hpp>
+
+#include <iostream>
+#include <string>
+
+#include <Core/Animation/KeyFramedValueController.hpp>
+#include <Core/Asset/FileData.hpp>
+#include <Core/Math/Math.hpp>
+#include <Core/Tasks/Task.hpp>
+#include <Core/Tasks/TaskQueue.hpp>
+
+#include <Engine/FrameInfo.hpp>
+#include <Engine/RadiumEngine.hpp>
+#include <Engine/Scene/SkeletonComponent.hpp>
+#include <Engine/Scene/SkinningComponent.hpp>
+
+using namespace Ra::Core::Animation;
+
+namespace Ra {
+namespace Engine {
+namespace Scene {
+
+SkeletonBasedAnimationSystem::SkeletonBasedAnimationSystem() : System(), m_xrayOn( false ) {}
+
+// System Interface
+
+void SkeletonBasedAnimationSystem::generateTasks( Core::TaskQueue* taskQueue,
+                                                  const FrameInfo& frameInfo ) {
+    for ( auto compEntry : m_components )
+    {
+        // deal with AnimationComponents
+        if ( auto animComp = dynamic_cast<SkeletonComponent*>( compEntry.second ) )
```

Comment:
So if I understand well, dynamic_cast are "kind of" authorized.
Since the goal here is only to separate `SkeletonComponent` from `SkinningComponent`, would it be nicer to have `m_components` registering one kind, and another `Component` list registering the other kind?

Reply:
yes, `dynamic_cast` can be used in a few situations and there is nothing bad about it if it is not too frequent (in which case, there is a design problem)
Your proposal is an option, yes, and deserve to be examine. But `m_component` is inherited from `System` class, this might have an impact on the way components managed by a system can be accessed, mainly for "uniformity" access to the components.

Reply:
Yes you are right.
Maybe having two additional lists, one per kind of component? But that's redundant data (only pointers), made to avoid a `dynamic_cast`. Not sure it's better.